framework/src/base/ComputeResidualThread.C

```diff
@@ -91,6 +91,24 @@ ComputeResidualThread::subdomainChanged()
     }
   }
 
+  // Interface DG Kernel dependencies
+  for (std::set<unsigned int>::const_iterator id_it = subdomain_boundary_ids.begin();
+      id_it != subdomain_boundary_ids.end();
+      ++id_it)
+  {
+    std::vector<DGKernel *> interface_dgks;
+    _sys.getDGKernelWarehouse(_tid).activeInterfaceDG(*id_it, interface_dgks);
```

Comment:
@andrsd 
I modeled this after an earlier version of the onBoundary method, which had this:

  std::vector<IntegratedBC *> bcs;
  _sys.getBCWarehouse(_tid).activeIntegrated(bnd_id, bcs);

It looks like the onBoundary method has been updated to look like the other methods (e.g. returning a reference to a container like you say), so I will fix my method to be consistent with the rest.